File: leetCode/fukun/201-300/lc_206.js
function ListNode(val) {
    this.val = val;
    this.next = null;
}

const reverseList = function (head) {
    let tmp = null;
    while (head) {
        let next = head.next;
        head.next = tmp;
        tmp = head;
        head = next;
    }
    return tmp;
};

let ln0 = new ListNode(0);
let ln1 = new ListNode(1);
let ln2 = new ListNode(1);
ln0.next = ln1;
ln1.next = ln2;
console.log(reverseList(ln0));

File: leetCode/lwc/0201-0300/lc_206.java
package solution2;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/**
 * @author lwc
 * @date 2019/10/21 22:30
 */
public class U206 {
    public ListNode reverseList(ListNode head) {
        if(head == null || head.next == null) return head;
        ListNode cur = head;
        while(cur.next.next != null){
            cur = cur.next;
        }
        ListNode preHead = cur.next;
        cur.next = null;
        preHead.next = reverseList(head);
        return preHead;
    }

    public ListNode reverseList2(ListNode head) {
        if(head == null || head.next == null) return head;

        int pos = 1;
        List<ListNode> list = new ArrayList<>();
        list.add(head);
        list.add(head.next);
        ListNode cur = head.next;
        head.next = null;

        while(cur != null){
            list.add(cur.next);
            cur.next = list.get(pos-1);
            cur = list.get(++pos);
        }

        return list.get(pos-1);
    }

}
